beacon_node/lighthouse_network/src/peer_manager/peerdb/score.rs

```diff
@@ -39,11 +40,68 @@ const GOSSIPSUB_NEGATIVE_SCORE_WEIGHT: f64 =
     (MIN_SCORE_BEFORE_DISCONNECT + 1.0) / GOSSIPSUB_GREYLIST_THRESHOLD;
 const GOSSIPSUB_POSITIVE_SCORE_WEIGHT: f64 = GOSSIPSUB_NEGATIVE_SCORE_WEIGHT;
 
+pub(crate) const MAX_STORED_PENALTY_RECORDS: usize = 20;
+
+#[derive(Clone, Debug, Serialize)]
+pub struct PenaltyRecord {
+    /// The action that caused the penalty
+    action: PeerAction,
+    /// Where the penalty came from
+    source: ReportSource,
+    /// The penalty message
+    msg: String,
+    /// The result of the penalty
+    result: ScoreUpdateResult,
+    /// The time when the penalty occured in unix millis
+    time_stamp: Option<u128>
+}
+
+impl PenaltyRecord {
+    /// Create a new penalty record
+    pub fn new(action: PeerAction, source:ReportSource, msg: impl Into<String>, result: ScoreUpdateResult) -> PenaltyRecord {
+        let time_stamp: Option<u128> = SystemTime::now()
+            .duration_since(UNIX_EPOCH).ok()
+            .map(|dur| dur.as_millis());
```

Comment:
Default to 0 so you don't need an option. Not critical and a timestamp of 0 is as obvious as a None